src/sentences/alm.rs

```diff
@@ -0,0 +1,224 @@
+use core::{fmt::Debug, ops::RangeInclusive, str};
+use nom::{
+    character::{
+        complete::{char, hex_digit0},
+        streaming::hex_digit1,
+    },
+    combinator::{map_res, opt},
+    IResult,
+};
+
+use crate::{Error, NmeaSentence, SentenceType};
+
+use super::utils::number;
+
+/// ALM - GPS Almanac Data
+///
+/// <https://gpsd.gitlab.io/gpsd/NMEA.html#_alm_gps_almanac_data>
+/// ```text
+///         1   2   3  4   5  6    7  8    9    10     11     12     13     14  15   16
+///         |   |   |  |   |  |    |  |    |    |      |      |      |      |   |    |
+///  $--ALM,x.x,x.x,xx,x.x,hh,hhhh,hh,hhhh,hhhh,hhhhhh,hhhhhh,hhhhhh,hhhhhh,hhh,hhh*hh<CR><LF>
+/// ```
+///  
+///  Field Number:
+///  
+///  1. Total number of messages
+///  2. Sentence Number
+///  3. Satellite PRN number (01 to 32)
+///  4. GPS Week Number (range 0 to 2^13 - 1), where:
+///     - 0 is the week of the GPS Week Number epoch on January 6th 1980;
+///     - 8191 is the week that precedes the next rollover on January 6th 2137;
+///  6. Eccentricity
+///  7. Almanac Reference Time
+///  8. Inclination Angle
+///  9. Rate of Right Ascension
+/// 10. Root of semi-major axis
+/// 11. Argument of perigee
+/// 12. Longitude of ascension node
+/// 13. Mean anomaly
+/// 14. F0 Clock Parameter
+/// 15. F1 Clock Parameter
+/// 16. Checksum
+///  
+///  Fields 5 through 15 are dumped as raw hex.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct AlmData {
+    total_number_of_messages: Option<u16>,
+    sentence_number: Option<u16>,
+    satellite_prn_number: Option<u8>,
+    gps_week_number: Option<u16>,
```

Comment:
be explicit and document that we're talking about a 13-bit week number support (the newer) and add a method for calculating the 10-bit representation (if the user is using older hardware)
- [ ] Document field
- [ ] add method for returning the 10-bit week number